applications/ALEapplication/python_scripts/ale_analysis.py

```diff
@@ -0,0 +1,283 @@
+from __future__ import print_function, absolute_import, division  # makes KratosMultiphysics backward compatible with python 2.6 and 2.7
+
+# Importing Kratos
+import KratosMultiphysics
+import KratosMultiphysics.ALEApplication as KratosALE
+
+# Importing the solvers (if available)
+try:
+    import KratosMultiphysics.ExternalSolversApplication
+    KratosMultiphysics.Logger.PrintInfo("ExternalSolversApplication", "succesfully imported")
+except ImportError:
+    KratosMultiphysics.Logger.PrintInfo("ExternalSolversApplication", "not imported")
+try:
+    import KratosMultiphysics.EigenSolversApplication
+    KratosMultiphysics.Logger.PrintInfo("EigenSolversApplication", "succesfully imported")
+except ImportError:
+    KratosMultiphysics.Logger.PrintInfo("EigenSolversApplication", "not imported")
+
+
+class ALEAnalysis(object): # TODO in the future this could derive from a BaseClass in the Core
+    """
+    This class is the main-script of the ALEapplication put in a class
+
+    It can be imported and used as "black-box"
+    """
+    def __init__(self, ProjectParameters, external_model_part=None):
+        if (type(ProjectParameters) != KratosMultiphysics.Parameters):
+            raise Exception("Input is expected to be provided as a Kratos Parameters object")
+        self.ProjectParameters = ProjectParameters
+        self.__CreateSolver(external_model_part)
+
+    #### Public functions to run the Analysis ####
+    def Run(self):
+        self.Initialize()
+        self.RunMainTemporalLoop()
+        self.Finalize()
+
+    def RunMainTemporalLoop(self):
+        while self.time < self.end_time:
```

Comment:
Its just an opinion : ALEAnalyis cannot be used as a stand alone solver as I understand it. So I would rather put RunMainTemporalLoop and Run functions to be pass. This can avoid complications that arise later on with this analysis when these functions are used.

Reply:
it CAN be used as a standalone solver (e.g. if one just wants to move the mesh)

Reply:
Yes ... I mean to say it should always be used in combination of other physics solvers  .. consider the case we run a temporal loop only with the mesh motion solver ... ! Since it is to be used in combination with other physics solvers, it will not need a temporal loop, and the functions I mentioned.

Reply:
I don't think it needs to be used together with a physical solver. In fact, the nice extra feature of such a "black-box" analysis script is, that you can use it standalone. The temporal loop could be anything, think of a time-dependend precsribed motion pattern / mesh update. However, I see that in cases when it is only used as mesh-motion solver, the wording ALEAnalysis is not descriptive anymore.